lib/r10k/api.rb

```diff
@@ -211,8 +243,74 @@ def resolve_environment(env_map)
     #
     # @param module_name [String] Name of the module to be resolved, should match the value of the "name" key in the supplied environment map.
     # @param env_map [Hash] A hashmap representing a single environment's desired state.
+    # @param opts [Hash] Additional options as defined.
+    # @option opts [String] :cachedir Base path where caches are stored.
     # @return [Hash] A copy of env_map with a new :resolved_version key/value pair added for the specified module.
-    def resolve_module(module_name, env_map)
+    # @raise [RuntimeError, R10K::API::UnresolvableError] Something bad happened.
+    def resolve_module(module_name, env_map, opts={})
+      mod_found = false
+
+      env_map[:modules].map! do |mod|
+        if mod[:name] == module_name
+          case mod[:type].to_sym
+          when :git
```

Comment:
yes, I reached that conclusion as well :) I'll try to get that done in the morning once I've got the unit tests written